src/src/pages/Experiment/components/ExperimentSelectionPopover/ExperimentSelectionPopover.tsx

```diff
@@ -0,0 +1,260 @@
+      return `${name.slice(0, 27)}...${name.slice(-26)}`;
+    }
+    return name;
+  }
+
+  function handleExperimentClick(experimentName: string): void {
+    onSelectExperimentNamesChange(experimentName);
+    // TODO: Figure out a better way to re-render
+    forceUpdate();
+  }
+
+  function experimentInList(
+    experimentName: string,
+    selectedExperimentNames: string[],
+  ): boolean {
+    return selectedExperimentNames.includes(experimentName);
+  }
+
+  function handleSearchInputChange(
+    e: React.ChangeEvent<HTMLInputElement>,
+  ): void {
+    setSearchValue(e.target.value);
+
+    if (isRegexSearch) {
+      try {
+        const regex = new RegExp(e.target.value, 'i');
+        setInvalidRegex(false);
+        const options = experimentsData?.filter((experiment) =>
+          regex.test(experiment.name),
+        );
+        setVisibleExperiments(options || []);
+      } catch (error) {
+        setInvalidRegex(true);
+      }
+    } else {
+      const options = experimentsData?.filter((experiment) =>
+        experiment.name.toLowerCase().includes(e.target.value.toLowerCase()),
+      );
+      setVisibleExperiments(options || []);
+    }
+  }
+
+  function toggleAllExperiments(checked: boolean): void {
+    const visibleExperimentNames = visibleExperiments?.map(
+      (experiment) => experiment.name,
+    );
+    // If all experiments are selected, deselect all
+    // otherwise, select all that are unselected
+    if (checked) {
+      onToggleAllExperiments(visibleExperimentNames);
+    } else {
+      const unselectedExperiments = visibleExperimentNames?.filter(
+        (experimentName) => !selectedExperimentNames.includes(experimentName),
+      );
+      onToggleAllExperiments(unselectedExperiments);
+    }
+  }
+
+  function allExperimentsSelected(): boolean {
+    return (
+      visibleExperiments.length > 0 &&
+      visibleExperiments.every((experiment) =>
+        selectedExperimentNames.includes(experiment.name),
+      )
+    );
+  }
+
+  return (
+    <ErrorBoundary>
+      <div className='ExperimentSelectionPopover'>
+        <div className='ExperimentSelectionPopover__headerContainer'>
+          <div className='ExperimentSelectionPopover__headerContainer__titleContainer'>
+            <Text size={14} tint={100} weight={700}>
+              Experiments
+            </Text>
+          </div>
+        </div>
+        <div className='ExperimentSelectionPopover__contentContainer'>
+          <div className='ExperimentSelectionPopover__contentContainer__experimentsListContainer'>
+            <div className='ExperimentSelectionPopover__contentContainer__experimentsListContainer__experimentList ScrollBar__hidden'>
+              <div
+                className={
+                  invalidRegex
+                    ? 'ExperimentSelectionPopover__searchContainer error'
+                    : 'ExperimentSelectionPopover__searchContainer'
+                }
+              >
+                <Checkbox
```

Comment:
maybe a tooltip on this checkbox, "Select all visible"?

Reply:
Done!